### DISCUSSION TITLE
Windows vs Linux compiler performance

### DISCUSSION THREAD

**Discussion by dennisma** [earlephilhower/arduino-pico#2228] [Q&A] [answered]
Noticed that the Earle Philhower Arduino core for RP2040 compiles "much" faster on Linux machines than on Windows (2x - 3x faster). 

We have a number of older PC's in a classroom setting running Win 10 and the compilation process  terribly slow.   I can run twice as fast on an 8Gb Raspberry Pi 5 than an older Intel I5 (the classroom pc).

I also tested this out on an I9 laptop with 64Gb running Win 11 and running WSL with Arduino installed on Linux running side by side with arduino installed with Windows.    Compilation time under Ubuntu under WSL is significantly faster for rp2040 as well.

(**note:** this is just compile times... not uploading)

So my question is... is there a way to make the Windows compilation time run faster?

**Comment by earlephilhower** [comment, marked as answer]
Could it be antivirus related?  GCC will touch 100s of files per compilation and if MS Defender/Avast/etc. is scanning through them it will burn CPU like mad.  Is there a way to disable antivirus (just for testing!) and compare times?  If that gives back a good chunk, then maybe there are special directory or file type exclusions you can set?

It could just be a function of the MinGW runtime, which is used to allow GCC to be built as a native app.  In that case, I think you're kind of out of luck. 

Really bodge-tastic, but it is also possible to run VNC server under WSL and start up the Linux Arduino.  Use VNCViewer to actually display and interact with it.  I believe the whole NTFS filesystem is available so you can still access files normally.

**Reply by dennisma** [reply]
Yes I had already disabled Defender... it was the first thing.  It ran slightly faster but not that much more.    Arduino running inside WSL can compile perfectly fine but there are issues getting it to upload the code.   I tried using the USBIPD-WIN service to allow it to share the USB port.   That does work for Serial communication just fine but uploading code is a bit 'tricky' given the process for going into UF2 mode (WSL loses the shared port everytime).

Anyway I believe your take in MinGW under Windows is correct... it was what prompted me to try running it under WSL/Ubuntu.   I'll look into it more and see if I can make it work easier.   

Some of our students are pretty young and we cannot make this too hard of a process  on them.   Worst case scenario we just stick with slow machines for now (or buy a bunch of Raspberry PI 5's   ;-) 

Thanks again for the suggestions.
